src/utils/concatenate_sentence.py

```diff
@@ -0,0 +1,166 @@
+    origin_target_data = [pair[target_lang] for pair in data["train"]["translation"]]
+    print(len(origin_source_data))
+
+    # if back translate is true, back translate the origin data else
+    # set the pseudo data to be the same as the origin data
+    if do_back_translation:
+        pseudo_source_data, index_translated = back_translate(
+            origin_target_data,
+            back_translation_model,
+            back_translation_percent,
+        )
+        pseudo_target_data = [origin_target_data[index] for index in index_translated]
+    else:
+        pseudo_source_data = origin_source_data
+        pseudo_target_data = origin_target_data
+
+    # If sequential, concantenate all sentences in the source data with the sentence at
+    # the next index in the pseudo data. If it's not sequential then randomly select
+    # sentences to concatenate
+    if sequential:
+        origin_indexes = list(range(len(origin_source_data)))
+        pseudo_indexes = list(range(1, len(pseudo_source_data)))
+    else:
+        origin_indexes = random.sample(
+            range(len(origin_source_data)), number_of_concatenations
+        )
+        pseudo_indexes = random.sample(
+            range(len(pseudo_source_data)), number_of_concatenations
+        )
+
+    concantenated_source = []
+    concantenated_target = []
+
+    for origin_index, pseudo_index in list(zip(origin_indexes, pseudo_indexes)):
+        new_source = (
+            origin_source_data[origin_index]
+            + f" {seperator_token} "
+            + pseudo_source_data[pseudo_index]
+        )
+        # check if the length of the concatenated sentence meets the threshold
+        if len(new_source) >= sentence_length_threshold:
+            concantenated_source.append(new_source)
+            concantenated_target.append(
+                origin_target_data[origin_index]
+                + f" {seperator_token} "
+                + pseudo_target_data[pseudo_index]
+            )
+    new_source_data = origin_source_data + concantenated_source
+    new_target_data = origin_target_data + concantenated_target
+
+    return new_source_data, new_target_data
+
+
+if __name__ == "__main__":
+    # This the name of the subset of the dataset to use
+    # in the case of mafand, it's the language pair
+    SUBSET = "en-yor"
+    # separator token. This should ideally be the separator token used by
+    # the model you would train with the output data, not the back translation model.
+    SEPARATOR_TOKEN = "</s>"
+    SENTENCE_LENGTH_THRESHOLD = 10
+    SEQUENTIAL = False
+    DO_BACK_TRANSLATION = True
+    # In a case where you are doing back translation, this parameter should be less
+    # that back_translation_percent * <length of your data>, in this case 33.22(6644 * 0.005)
+    NUMBER_OF_CONCATENATION = 10
+    BACK_TRANSLATION_MODEL = "masakhane/mbart50_yor_en_news"
+    BACK_TRANSLATION_PERCENT = 0.005
```

Comment:
All these parameters should be available as variables that can be run with the script.